### DISCUSSION TITLE
Can Supervision do the following?

### DISCUSSION THREAD

**Discussion by PuckStar** [roboflow/supervision#298] [Q&A] [answered]
What an interesting project this is!

Before I dive into it I would like to know if someone can tell me if the following should be possible with this project:
In a video detect certain type of vehicles (car, bus, truck etc.) and registers (ideally write to a file) each time such vehicle follows a certain path in the video.

As example the video is aimed at an intersection. Cars, busses, trucks etc. drive there, but some go straight, some take a turn.
I'd like to automate that it's written to a file (as text) how many cars drove straight, how many cars took a turn left, took a turn right etc. And the same for the other type of vehicles.

Does anyone know if this should be possible?
Thanks, and let me know if anything is unclear.

**Comment by maddust** [comment, marked as answer]
hello @PuckStar looks like you are looking to do a Turning movement count. 

To answer your question. Yes is totally doable . I've have implement this in my project using supervision tools. 
here some guides to achieve it:

The first challenge is the current implementation of the lineZone Counter . the devs are working on a solution for this but there is currently  a "bug" that objects are being counted outside the linezone limits. 

Here is a high level overview of how I've  implement this : 
- Set lines or zones on each approach of the intersection 
- Determine on which direction the object passed (In or out) for each LineZone, you can determine this by the tracker state  of each object . if you are using PolygonZones you just need to know if the object was inside of it . 
- Append/save the results on a Database with timestamp, object ID, Classes CounterName and Direction Crossed 
- Then using the Database i made script to match ID Parings of the First and Last LineZone Counter Crossed by each object.  giving you the origin and destination of each object

I went even deeper on this and created a visualization tool to display the trajectories for each object and turns. 

Unfortunately i cannot share the entire code because is part of a closed project but hope this gives you some light how to implement it 

![ezgif com-video-to-gif](https://github.com/roboflow/supervision/assets/16510431/69ab4bd6-e667-416c-8d7f-3cd02b2cd64d)

**Reply by SkalskiP** [reply]
@maddust this looks amazing! I think you can do it with `PolygonZone` as well. The question could you share some raw video from that street? I'd love to use it in my upcoming YouTube video.

**Reply by maddust** [reply]
@SkalskiP thanks! all kudos too supervision .  Thats correct it can be done with PolygonZone as well. 

regarding the video absolutely here you go . hope it helps . i uploaded an extra one just in case in the drive 
[https://drive.google.com/drive/folders/1QGUI49swrGj_ykDzXMyn7fONU0kI3vyu?usp=sharing](url)

**Reply by SkalskiP** [reply]
Hi, @maddust 👋🏻! I'm starting my work on the traffic analysis tutorial today. You cure I can use those videos in YouTube video?

**Reply by maddust** [reply]
@SkalskiP yes , feel free to use it. i own and recorded both videos

**Comment by PuckStar** [comment]
@maddust thanks a lot for your message.
Very nice to hear that it should be possible.

I'm not so tech savvy so not sure how to even start to get this working but perhaps the tutorial of @SkalskiP might help.
Looking forward to it!

If I can't manage myself I'm even willing to pay someone to help me out.
